### PR TITLE
adds sonar-scanner-cli-entrypoint and compat packages

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: sonar-scanner-cli.yaml [hunk 1]
```diff
@@ -37,8 +37,24 @@ pipeline:
...
+subpackages:
+  - name: sonar-scanner-cli-compat
+    dependencies:
+      runtime:
+        - sonar-scanner-cli
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/opt/sonar-scanner/bin
+          mkdir -p "${{targets.subpkgdir}}"/opt/sonar-scanner/lib
+          ln -s /usr/bin/sonar-scanner-cli "${{targets.subpkgdir}}"/opt/sonar-scanner/bin/
```
Comment (jamonation) [inline sonar-scanner-cli.yaml:55]:
```suggestion
          ln -s /usr/bin/sonar-scanner-cli "${{targets.subpkgdir}}"/opt/sonar-scanner/bin/
          ln -s /usr/bin/sonar-scanner-cli "${{targets.subpkgdir}}"/opt/sonar-scanner/bin/sonar-scanner # entrypoint.sh script expects this

```

File: sonar-scanner-cli-entrypoint.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      expected-sha256: d93310f2a96b766e0c96de33b519d0e414d863e28a433d2ca7832fcaa5926979
+      extract: false
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/bin/
+      chmod +x entrypoint.sh
+      cp entrypoint.sh ${{targets.destdir}}/usr/bin/
+
+update:
+  enabled: true
```
Comment (jamonation) [inline sonar-scanner-cli-entrypoint.yaml:30]:
```suggestion
  manual: true
```

There's no obvious way with tags to automate updates.

File: sonar-scanner-cli-entrypoint.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  contents:
+    packages:
+      - busybox
+
+pipeline:
+  - uses: fetch
+    with:
+      uri: https://raw.githubusercontent.com/SonarSource/sonar-scanner-cli-docker/${{package.version}}/bin/entrypoint.sh
+      expected-sha256: d93310f2a96b766e0c96de33b519d0e414d863e28a433d2ca7832fcaa5926979
+      extract: false
```
Comment (jamonation) [inline sonar-scanner-cli-entrypoint.yaml:22]:
```suggestion
pipeline:
  - uses: git-checkout
    with:
      repository: https://github.com/SonarSource/sonar-scanner-cli-docker
      expected-commit: 8ee21641b014850a1c50d0635aa337bf9a6ba369
      tag: ${{package.version}}
```

File: sonar-scanner-cli-entrypoint.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/bin/
+      chmod +x entrypoint.sh
+      cp entrypoint.sh ${{targets.destdir}}/usr/bin/
```
Comment (jamonation) [inline sonar-scanner-cli-entrypoint.yaml:27]:
```suggestion
      install -m755 entrypoint.sh ${{targets.destdir}}/usr/bin/
```

**Review by jamonation — CHANGES_REQUESTED** [review]
Couple small changes and then this looks good

File: sonar-scanner-cli-entrypoint.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  - uses: git-checkout
+    with:
+      repository: https://github.com/SonarSource/sonar-scanner-cli-docker
+      expected-commit: 8ee21641b014850a1c50d0635aa337bf9a6ba369
+      tag: ${{package.version}}
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/bin/
+      chmod +x entrypoint.sh
+      install -m755 entrypoint.sh ${{targets.destdir}}/usr/bin/
```
Comment (jamonation) [inline sonar-scanner-cli-entrypoint.yaml:27]:
```suggestion
      install -m755 bin/entrypoint.sh ${{targets.destdir}}/usr/bin/
```

File: sonar-scanner-cli-entrypoint.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Fetches the sonar-scanner-cli entrypoint script used in the upstream docker image. We
+# need this for our image, but as it's stored in a different repo vs the source
+# code, we need to fetch it and be able to auto-update.
+package:
+  name: sonar-scanner-cli-entrypoint
+  version: 10.0.3.1430_5.0.1
```
Comment (jamonation) [inline sonar-scanner-cli-entrypoint.yaml:6]:
```suggestion
  version: "10.0.3.1430_5.0.1"
```

File: sonar-scanner-cli-entrypoint.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+environment:
+  contents:
+    packages:
+      - busybox
+
+var-transforms:
+  - from: ${{package.version}}
+    match: '^(\d+\.\d+\.\d+\.\d+)\.(\d+\.\d+\.\d+)$'
+    replace: '$1_$2'
+    to: mangled-package-version
```
Comment (jamonation) [inline sonar-scanner-cli-entrypoint.yaml:21]:
```suggestion
```

File: sonar-scanner-cli-entrypoint.yaml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/SonarSource/sonar-scanner-cli-docker
+      expected-commit: 8ee21641b014850a1c50d0635aa337bf9a6ba369
+      tag: ${{vars.mangled-package-version}}
```
Comment (jamonation) [inline sonar-scanner-cli-entrypoint.yaml:28]:
```suggestion
```

**Review by jamonation — CHANGES_REQUESTED** [review]
This should build!

File: sonar-scanner-cli-entrypoint.yaml [hunk 9]
```diff
@@ -0,0 +1,30 @@
+# Fetches the sonar-scanner-cli entrypoint script used in the upstream docker image. We
```
Comment (jamonation) [inline sonar-scanner-cli-entrypoint.yaml:1]:
```suggestion
#nolint:valid-pipeline-git-checkout-tag
# Fetches the sonar-scanner-cli entrypoint script used in the upstream docker image. We
```

**Review by EyeCantCU — APPROVED** [review]

**Review by jamonation — APPROVED** [review]

**Pull request merged**